_assets/js/carousel.js

```diff
@@ -0,0 +1,335 @@
+  }
+
+  // Helper function: Test if element has a specific class
+  function hasClass(el, className) {
+    if (el.classList) {
+      return el.classList.contains(className);
+    } else {
+      return new RegExp('(^| )' + className + '( |$)', 'gi').test(el.className);
+    }
+  }
+
+  // Initialization for the carousel
+  // Argument: set = an object of settings
+  // Possible settings:
+  // id <string> ID of the carousel wrapper element (required).
+  // slidenav <bool> If true, a list of slides is shown.
+  // animate <bool> If true, the slides can be animated.
+  // startAnimated <bool> If true, the animation begins
+  // immediately.
+  // If false, the animation needs
+  // to be initiated by clicking
+  // the play button.
+  function init(set) {
+    // Make settings available to all functions
+    settings = set;
+
+    // Select the element and the individual slides
+    carousel = document.getElementById(settings.id);
+
+    if (!carousel) return;
+
+    slides = carousel.querySelectorAll('.slide');
+
+    carousel.className = 'active carousel';
+
+    const ctrls = document.createElement('ul');
+
+    ctrls.className = 'controls';
+    ctrls.innerHTML =
+      '<li>' +
+      '<button type="button" class="btn-prev" alt="Previous Item" />' +
+      '</li>' +
+      '<li>' +
+      '<button type="button" class="btn-next" alt="Next Item" />' +
+      '</li>';
+
+    ctrls.querySelector('.btn-prev').addEventListener('click', function () {
+      prevSlide(true);
+    });
+    ctrls.querySelector('.btn-next').addEventListener('click', function () {
+      nextSlide(true);
+    });
+
+    carousel.appendChild(ctrls);
+    // If the carousel is animated or a slide navigation is requested in the settings, another unordered list that contains those elements is added. (Note that you cannot supress the navigation when it is animated.)
+    if (settings.slidenav || settings.animate) {
+      slidenav = document.createElement('ul');
+
+      slidenav.className = 'slidenav';
+
+      if (settings.animate) {
+        const li = document.createElement('li');
+
+        if (settings.startAnimated) {
+          li.innerHTML = '<button data-action="stop" aria-label="Stop Animation">￭</button>';
+        } else {
+          li.innerHTML = '<button data-action="start" aria-label="Start Animation">▶</button>';
+        }
+
+        slidenav.appendChild(li);
+      }
+
+      if (settings.slidenav) {
+        forEachElement(slides, function (el, i) {
+          let li = document.createElement('li');
+          let klass = i === 0 ? 'class="current" ' : '';
+
+          li.innerHTML = '<button ' + klass + 'data-slide="' + i + '"/>';
+          slidenav.appendChild(li);
+        });
+      }
+
+      slidenav.addEventListener(
+        'click',
+        function (event) {
+          const button = event.target;
+          if (button.localName == 'button') {
+            if (button.getAttribute('data-slide')) {
+              if (settings.startAnimated) {
```

Comment:
Use guard clauses to flatten this code.

It's hard to visually tie the ifs to the else ifs, so [it's hard to see bugs like this](https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html) which might appear